src/main/resources/assets/computercraft/lua/rom/apis/textutils.lua

```diff
@@ -149,10 +158,32 @@ local function tabulateCommon( bPaged, ... )
 end
 
 function tabulate( ... )
+    tArgs = { ... }
+    for i=1,#tArgs,2 do
```

Comment:
The checks in tabulate / pagedTabulate could simply be performed into tabulateCommon with an error level of 3.

Moving them there would also allow use of [arg.n](https://www.lua.org/pil/5.2.html), avoiding an extra function call (you can only refer to either ... or arg within a vararg function: not both). Downside is that's broken after Lua 5.1.